sycl/include/sycl/ext/oneapi/bfloat16.hpp

```diff
@@ -61,7 +61,9 @@ class bfloat16 {
 #if defined(__SYCL_DEVICE_ONLY__)
 #if defined(__NVPTX__)
 #if (__SYCL_CUDA_ARCH__ >= 800)
-    return __nvvm_f2bf16_rn(a);
+    detail::Bfloat16StorageT res;
+    asm volatile("cvt.rn.bf16.f32 %0,%1;" : "=h"(res) : "f"(a));
```

Comment:
why `asm` here, and `__asm__` elsewhere in this patch?

Reply:
opencl c needs "__asm__" it doesn't know about "asm".

Reply:
given that `asm` is an extension, why not use the `__asm__` keyword everywhere?

Reply:
asm seems to be the standard in clang. I've followed this. asm is also consistent with the nvptx docs. Do you think it matters?

Reply:
Not really. It just looked a little weird having two versions of them in the same patch, but your rationale makes sense